apps/coordinator/src/components/Wallet/TransactionsTab/hooks.ts

```diff
@@ -18,6 +21,211 @@ import { useGetClient } from "../../../hooks/client";
  * For now, we're keeping it simple and just showing what's pending.
  */
 
+/**
+ * ==============================================================================
+ * TRANSACTION VALUE CALCULATION HELPERS
+ * ==============================================================================
+ */
+
+/**
+ * Check if a transaction output belongs to the wallet
+ */
+const belongsToWallet = (
+  output: { scriptPubkeyAddress?: string },
+  walletAddresses: string[],
+): boolean => {
+  const outputAddress = output.scriptPubkeyAddress;
+  return Boolean(outputAddress && walletAddresses.includes(outputAddress));
+};
+
+/**
+ * Helper function to check if we have complete input data for a transaction
+ */
+const hasCompleteInputData = (tx: any): boolean => {
+  if (!tx.vin || !Array.isArray(tx.vin)) return false;
+
+  return tx.vin.every(
+    (input: any) =>
+      input.prevout && input.prevout.scriptpubkey_address !== undefined,
+  );
+};
+
+/**
+ * Calculate transaction value from transaction details (private client)
+ * This gives the most accurate calculation as bitcoind provides detailed category information and amount field
+ */
+const calculateValueFromDetails = (details: any[]): number => {
+  return details.reduce((valueToWallet, detail) => {
+    const amountInSats = Number(bitcoinsToSatoshis(detail.amount));
+
+    if (
+      detail.category === "receive" ||
+      detail.category === "generate" ||
+      detail.category === "immature" ||
+      detail.category === "send" // Amount is already negative for send
+    ) {
+      return valueToWallet + amountInSats;
+    }
+
+    return valueToWallet;
+  }, 0);
+};
+
+/**
+ * Convert output value to satoshis, handling both string and number formats
+ */
+const outputValueToSatoshis = (value: any): number => {
+  if (typeof value === "undefined" || value === null) return 0;
+  return Number(bitcoinsToSatoshis(value));
+};
```

Comment:
I think we can make this more concise and better typed. Something like this
```suggestion
const outputValueToSatoshis = (value?: number | string): number => 
  bitcoinsToSatoshis(value ?? 0);
```

Reply:
Thanks for the feedback changes [done](https://github.com/caravan-bitcoin/caravan/pull/225/commits/e64fa4d2765afe975bb0b7a0abf69c8ceca9abb5)